mac/KeymanEngine4Mac/KeymanEngine4Mac/KME/KMEngine.m

```diff
@@ -86,18 +86,41 @@ - (NSArray *)processEvent:(NSEvent *)event {
 }
 
 - (NSArray *)processGroup:(KMCompGroup *)gp event:(NSEvent *)event {
-    NSArray *actions = nil;
-    
     if (!gp)
         return nil;
+    
+    if (![event respondsToSelector:@selector(keyCode)])
+        return nil;
 
     if (([event modifierFlags] & NSEventModifierFlagCommand) == NSEventModifierFlagCommand)
         return nil; // Engine should NEVER attempt to process characters when the Command key is pressed.
+    
+    @try {
+        return [self processGroupInternal: gp event:event];
+    }
+    @catch (NSException *e) {
+        NSLog(@"Error = %@", e.description);
+        return nil;
```

Comment:
I don't generally like masking exceptions like this; they tend to cause ripple effect issues down the line. I can see you are logging it but it's not very accessible to end users. What's the rationale for the catch-all?

Reply:
The actual exception I want to catch is internal, so apparently I can't trap it. I *might* be able to figure out of there is a more narrowly defined superclass for the exception I want. If not, maybe I can just wrap the individual calls I'm concerned about. Another option might be to just ask the event its type and assume that if it is of the correct type(s) that it will be able to supply the correct information. If an exception is thrown from within the Engine it will probably knock Keyman out of commission, but maybe that is preferable (so we can find out and track it down).